### DISCUSSION TITLE
Usage of number_format() with TextFields

### DISCUSSION THREAD

**Discussion by theo-1** [filamentphp/filament#10975] [Help]
So I am using number_format function to display amount money value fields in my repeater. However, to do math operations I str_replace to get rid of commas them cast to float. Then go back the other way to display the results. 

It all seems very clumsy and by this post I'm asking if I'm doing things right or there is a smarter way. 

I would have thought there could be an internal raw representation of the numeric values different from how its presented: just like you can have a field name different from the label that's actually displayed.

**Comment by leandrocfe** [comment]
Maybe using this
https://filamentphp.com/docs/3.x/panels/getting-started#casting-the-price-to-an-integer

or packages like this?
https://github.com/akaunting/laravel-money

**Reply by theo-1** [reply]
Thanks so much
